packages/duckdb-wasm/src/bindings/udf_runtime.ts

```diff
@@ -14,36 +14,212 @@ function storeError(mod: DuckDBModule, response: number, message: string) {
     mod.HEAPF64[(response >> 3) + 2] = heapArray.byteLength;
 }
 
+function type_size(ptype: string) {
+    switch(ptype) {
+        case 'UINT8':
+        case 'INT8':
+            return 1;
+        case 'INT32':
+        case 'FLOAT':
+            return 4;
+        case 'INT64':
+        case 'UINT64':
+        case 'DOUBLE':
+        case 'VARCHAR':
+            return 8;
+        default:
+            return 0;
+    }
+}
+
+function ptr_to_arr(mod: DuckDBModule, ptr: number, ptype: string, n:number) {
+    const in_buf = mod.HEAPU8.subarray(ptr, ptr + n * type_size(ptype));
+
+    switch(ptype) {
+        case 'UINT8': {
+            return new Uint8Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'INT8': {
+            return new Int8Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'INT32': {
+            return new Int32Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'INT64': {
+            return new BigInt64Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'UINT64': {
+            return new BigUint64Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'FLOAT': {
+            return new Float32Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'DOUBLE': {
+            return new Float64Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        case 'VARCHAR': {
+            return new Float64Array(in_buf.buffer, in_buf.byteOffset, n);
+        }
+        default:
+            return new Array<string>(0); // cough
+    }
+}
+
+// this is called from webdb.cc/CallScalarUDFFunction, changes here need to be matched there
 export function callScalarUDF(
     runtime: DuckDBRuntime,
     mod: DuckDBModule,
     response: number,
     funcId: number,
-    bufferPtr: number,
-    bufferSize: number,
+    descPtr: number,
+    descSize: number,
+    ptrsPtr: number,
+    ptrsSize: number,
 ) {
     try {
         const udf = runtime._udfFunctions.get(funcId);
         if (!udf) {
-            storeError(mod, response, 'unknown udf with id: ' + funcId);
+            storeError(mod, response, 'Unknown UDF with id: ' + funcId);
+            return;
+        }
+
+        // schema description as json
+        const desc_str = new TextDecoder().decode(mod.HEAPU8.subarray(descPtr, descPtr + descSize));
+        const desc = JSON.parse(desc_str);
+        //console.log(desc);
+
+        // array of buffer pointers referred to from schema
+        const ptrs_buf = mod.HEAPU8.subarray(ptrsPtr, ptrsPtr + ptrsSize);
+        const ptrs = new BigUint64Array(ptrs_buf.buffer, ptrs_buf.byteOffset, ptrsSize / 8);
```

Comment:
We might want to change this one later.
https://caniuse.com/?search=BigUint64Array

We do a platform check for BigIntArrays in the remaining API.
https://github.com/duckdb/duckdb-wasm/blob/d050191e1cad0525b40d466d714610543371739f/packages/duckdb-wasm/src/platform.ts#L67

Reply:
Ah missed that one, will just switch them to Double, too, seems to work fine